### PR TITLE
HPCC-13026 Allow eclcc to do a better job of resourcing smaller clusters

### PR DESCRIPTION
**Pull request by richardkchapman**
Use target cluster to determine actual size.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Manual testing observing the tracing was as expected using -v

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: ecl/eclcc/eclcc.cpp [hunk 1]
```diff
@@ -2238,6 +2282,10 @@ int EclCC::parseCommandLineOptions(int argc, const char* argv[])
         else if (iter.matchFlag(optCheckDirty, "-checkDirty"))
         {
         }
+        else if (iter.matchOption(tempArg, "-cluster"))
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp]: trivial.  Can use optCluster directly in this line - no need to go via tempArg.

File: ecl/eclcc/eclcc.cpp [hunk 2]
```diff
@@ -257,11 +257,15 @@ class EclCC : public CInterfaceOf<ICodegenContextCallback>
 
     // interface ICodegenContextCallback
 
-    virtual void noteCluster(const char *clusterName);
-    virtual bool allowAccess(const char * category, bool isSigned);
+    virtual void noteCluster(const char *clusterName) override;
+    virtual void pushCluster(const char *clusterName) override;
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp]: This will clash with my PR for _TARGET_PLATFORM_

File: ecl/hql/hql.hpp [hunk 3]
```diff
@@ -204,6 +206,11 @@ interface ICodegenContextCallback : public IInterface
      * @param location      Location to use when reporting errors
      */
     virtual IHqlExpression *lookupDFSlayout(const char *filename, IErrorReceiver &errs, const ECLlocation &location, bool isOpt) const = 0;
+    /**
+     * Return number of nodes for the current dluster, via Dali lookup, or 0 if cannot be determined.
```
Comment (ghalliday) [inline ecl/hql/hql.hpp]: typo dluster->cluster

**Review by ghalliday — COMMENTED** [review]
A couple of minor comments.

**Comment by ghalliday** [comment]
@richardkchapman

**Pull request merged**